### PR TITLE
Convert more connection tests to check relaying behavior

### PR DESCRIPTION
**Pull request by akshayjshah**
These tests work without problems - they just needed to be converted to use `WithTestServer` instead of `WithVerifiedServer`.

I'm still working on a handful of other tests that currently don't work.

### CODE REVIEW

File: testutils/test_server.go [hunk 1]
```diff
@@ -28,8 +28,11 @@ import (
 	"testing"
 	"time"
 
+	"golang.org/x/net/context"
+
```
Comment (akshayjshah) [inline testutils/test_server.go]: Will fix import order before landing.

**Comment by prashantv** [comment]
The test failures were really hard to read (because some channels weren't closed, and include all channels in the output on failures). I rebased the muttley branch off dev to pick up #329. Can you rebase off that branch and push, if tests fail, hopefully we'll have better errors.

File: connection_test.go [hunk 2]
```diff
@@ -567,28 +577,33 @@ func TestReadTimeout(t *testing.T) {
...
 			assert.Equal(t, err, context.Canceled, "Call should fail due to cancel")
 		}
 	})
 	goroutines.VerifyNoLeaks(t, nil)
 }
 
 func TestWriteTimeout(t *testing.T) {
-	WithVerifiedServer(t, nil, func(ch *Channel, hostPort string) {
+	opts := testutils.NewOpts().SetRelay()
+
```
Comment (prashantv) [inline connection_test.go]: nit: delete blank to match other 1 line opts

**Comment by prashantv** [comment]
lgtm

**Comment by akshayjshah** [comment]
Landing, despite test flakiness.

**Pull request merged**